Libplanet.Net/Consensus/Context.cs

```diff
@@ -432,24 +428,6 @@ internal void AddMessage(ConsensusMessage message)
                 }
             }
 
-            if (message is ConsensusVote vote &&
-                (!vote.ProposeVote.Verify(vote.Remote!.PublicKey) ||
-                 !_validators.Contains(vote.ProposeVote.Validator)))
-            {
-                throw new InvalidValidatorVoteMessageException(
-                    "Received ConsensusVote message is made by invalid validator.",
-                    vote);
-            }
-
-            if (message is ConsensusCommit commit &&
-                (!commit.CommitVote.Verify(commit.Remote!.PublicKey) ||
-                 !_validators.Contains(commit.CommitVote.Validator)))
-            {
-                throw new InvalidValidatorVoteMessageException(
-                    "Received ConsensusCommit message is made by invalid validator.",
-                    commit);
-            }
-
```

Comment:
Previously the logic was depending on `Message.Remote`, which is actually peer information of the sender. As messages can be gossiped and transferred from another peer not the actual creator of the message, the whole logic became useless. I newly introduced new property `ConsensusMessage.Validator` which will replace `Message.Remote` in consensus, which can not be modified.

Reply:
I understood the useless of message validation logic on gossip, but still not understood about how to prevent fake message attack...
I think I can understand it after careful study on gossip logic, not for now.
So I'll resolve this conversation.